api/src/main/java/org/openmrs/module/eptsreports/reporting/utils/EptsReportUtils.java

```diff
@@ -72,4 +80,40 @@ public static void setupReportDefinition(ReportManager reportManager) {
 		ReportManagerUtil.setupReport(reportManager);
 	}
 	
+	/**
+	 * @param parameterizable
+	 * @param mappings
+	 * @param <T>
+	 * @return
+	 */
+	public static <T extends Parameterizable> Mapped<T> map(T parameterizable, String mappings) {
+		if (parameterizable == null) {
+			throw new NullPointerException("Programming error: missing parameterizable");
+		}
+		if (mappings == null) {
+			mappings = ""; // probably not necessary, just to be safe
+		}
+		return new Mapped<T>(parameterizable, ParameterizableUtil.createParameterMappings(mappings));
+	}
+	
+	/**
+	 * Adds a row to a dataset based on an indicator and a list of column parameters
+	 * 
+	 * @param cohortDsd the dataset
+	 * @param baseName the base columm name
+	 * @param baseLabel the base column label
+	 * @param indicator the indicator
+	 * @param columns the column parameters
+	 * @param columnNames the column names
+	 */
+	public static void addRow(CohortIndicatorDataSetDefinition cohortDsd, String baseName, String baseLabel,
+	        Mapped<CohortIndicator> indicator, List<ColumnParameters> columns, List<String> columnNames) {
+		int c = 0;
+		for (ColumnParameters column : columns) {
+			String name = baseName + "-" + (columnNames != null ? columnNames.get(c++) : column.getName());
```

Comment:
This if statement is not useful at the moment since columnNames is hardcoded with values and so is never null. 

I think we need to rework this bit of logic - it doesn't make sense to be defining an array with columnNames, when we already have a class which is meant to represent the column.

Reply:
I think this covered too. columnNames have been replaced by a property from ColumnParameters class.